src/Translumo.TTS/Engines/SileroTTSEngine.cs

```diff
@@ -0,0 +1,144 @@
+        Init();
+    }
+
+    private void Init()
+    {
+        _pythonEngine.Init();
+
+        _pythonEngine.Execute(() =>
+        {
+            var np = _pythonEngine.Import("numpy");
+            _ipython = _pythonEngine.Import("IPython");
+            dynamic torch = _pythonEngine.Import("torch");
+
+            using dynamic device = torch.device("cpu");
+            torch.set_num_threads(4);
+
+            _model = torch.package.PackageImporter(_modelPath).load_pickle("tts_models", "model");
+            _model.to(device);
+
+            _pyObjects.Add(_model);
+            _pyObjects.Add(np);
+            _pyObjects.Add(torch);
+            _pyObjects.Add(_ipython);
+        });
+
+        _voice = ((string[])_model.speakers).First();
+    }
+
+    public void SpeechText(string text)
+    {
+        _cancellationTokenSource.Cancel();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        Task.Factory.StartNew(() => GenerateAudio(text), _cancellationTokenSource.Token)
+            .ContinueWith((bytesTask) =>
+             {
+                 if (!bytesTask.IsCompletedSuccessfully)
+                 {
+                     return;
+                 }
+
+                 PlayWavBytes(bytesTask.Result);
+             }, _cancellationTokenSource.Token);
+    }
+
+    private byte[] GenerateAudio(string text) =>
+        _pythonEngine.Execute(() =>
+        {
+            using var audio = _model.apply_tts(text: text, speaker: _voice, sample_rate: _wave_rate);
+            using var pyAudio = _ipython.display.Audio(audio, rate: _wave_rate);
+            return (byte[])pyAudio.data;
+        });
+
+    private void PlayWavBytes(byte[] wavBytes)
+    {
+        _player.Stop();
+        using (var ms = new MemoryStream(wavBytes))
+        {
+            _player.Stream = ms;
+            _player.Play();
+        }
+    }
+
+    public void Dispose()
+    {
+        _cancellationTokenSource.Cancel();
+        _cancellationTokenSource.Dispose();
+        _player.Stop();
+        _player.Dispose();
+
+        foreach (var pyObject in _pyObjects)
+        {
+            pyObject.Dispose();
+        }
+
+        _pythonEngine.Dispose();
+    }
+
+    public static bool IsLanguageSupport(string langCode) => GetModelForLanguage(langCode) != null;
+
+    private string GetModelFullPath(string langCode)
+    {
+        var modelUrl = GetModelForLanguage(langCode)?.FileUrl
+            ?? throw new NotSupportedException($"{nameof(SileroTTSEngine)} not support language '{langCode}'");
+
+        var fileName = Path.GetFileName(modelUrl);
+        var fullPath = Path.Combine(
+            Global.ModelsPath,
+            _sileroModelDirectory,
+            fileName);
+
+        return fullPath;
+    }
+
+    private static ModelDescription? GetModelForLanguage(string lang) =>
+        lang switch
+        {
+            "en-US" =>
+                new("https://models.silero.ai/models/tts/en/v3_en.pt",
```

Comment:
need to update component archive (adding silero models) and bat files
v.0.8.5/_Components-v.0.8.0.zip
I've uploaded updated components archive and just silero models to draft release in my repo: https://github.com/igubanov/Translumo-TTS/releases/tag/untagged-e0fdfe5dadab698838e9

I didnt add functionalty to automatically download models

Reply:
Thanks for noticing! Yeah, I'm planning to reconsider artifacts storage model anyway